pkg/controller/dataimportcron-controller.go

```diff
@@ -734,49 +730,54 @@ func addDataImportCronControllerWatches(mgr manager.Manager, c controller.Contro
 }
 
 func (r *DataImportCronReconciler) cronJobExistsAndUpdated(ctx context.Context, cron *cdiv1.DataImportCron) (bool, error) {
-	var cronJob batchv1.CronJob
-	cronJobNamespacedName := types.NamespacedName{Namespace: r.cdiNamespace, Name: GetCronJobName(cron)}
-	if err := r.client.Get(ctx, cronJobNamespacedName, &cronJob); err != nil {
+	cronJob := batchv1.CronJob{}
+	cronJobKey := types.NamespacedName{Namespace: r.cdiNamespace, Name: GetCronJobName(cron)}
+	if err := r.client.Get(ctx, cronJobKey, &cronJob); err != nil {
 		return false, cc.IgnoreNotFound(err)
 	}
-	desired, err := r.newCronJob(cron)
-	if err != nil {
-		return false, err
-	}
-	current, err := sdk.StripStatusFromObject(&cronJob)
-	if err != nil {
-		return false, err
-	}
-	sdk.MergeLabelsAndAnnotations(desired, current)
```

Comment:
I see you got rid of all the SDK merging utils - are we ok with that?

Reply:
the concern is that the deepequal will be false most of the time due to status changes/omitted spec fields

Reply:
No nned for the SDK merging utils. Regarding the `DeepEqual` you were right. Fixed that.

Reply:
Would it make sense to DeepEqual the spec instead of the entire object? Or are the annotations/labels we are interested in?

Reply:
We DeepEqual the whole object as we are intersted also in annotations/labels which may update on new version. Our CronJobs are not reconciled that often, so it won't cost too much to be on the safe side.